lib/rules/camelcase.js

```diff
@@ -71,6 +76,11 @@ module.exports = {
         if (properties !== "always" && properties !== "never") {
             properties = "always";
         }
+        const ignorePattern = options.ignorePattern || null;
+
+        if (ignorePattern) {
+            config.ignorePattern = new RegExp(ignorePattern);
```

Comment:
Your idea sounds great. But if some pattern is set to ignorePattern, you need to check name parameter even if that pattern never mathches anything. It take more cost than doing null-checks.
What do you think?